temporalio/converter.py

```diff
@@ -0,0 +1,362 @@
+"""Base converter and default implementations for conversion to/from values/payloads."""
+
+import dataclasses
+import inspect
+import json
+from abc import ABC, abstractmethod
+from typing import Any, Optional, Tuple, Type
+
+import google.protobuf.json_format
+import google.protobuf.message
+import google.protobuf.symbol_database
+
+import temporalio.api.common.v1
+
```

Comment:
I would define an exception class here for users to throw (or use one of the already defined python exceptions) (e.g. [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) or `ValueError`)

Reply:
I think we can document that they can use `TypeError` or `ValueError` or their own `RuntimeError`

Reply:
Added documentation on proper errors to use in different situations.